packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -56,6 +57,11 @@ export interface IFunction extends IResource, ec2.IConnectable, iam.IGrantable {
    */
   readonly permissionsNode: ConstructNode;
 
+  /**
+   * The system architectures compatible with this lambda function.
+   */
+  readonly architecture?: Architecture;
```

Comment:
Can't we default this to `X86`?

Reply:
In the past it looks like we decided to intentionally not default this. See [this comment](https://github.com/aws/aws-cdk/pull/17121#discussion_r743838650) and [this comment](https://github.com/aws/aws-cdk/pull/17000#discussion_r739116360)